### DISCUSSION TITLE
Some suggestions

### DISCUSSION THREAD

**Discussion by DareckGobos** [BeansGalaxy/Beans-Backpacks-2#95] [Feedback]
The mod demonstrates impressive potential as a supplementary backpack for storage or extra inventory while maintaining Minecraft's Vanilla aesthetic. However, one aspect that could be improved is its versatility. Currently, the ability to open the backpack while wearing it or holding it in hand is lacking. Implementing this feature would enhance its functionality and usability. Additionally, customizable templates further enhance its appeal. Overall, commendable work, with room for enhancement.

Could the idea of implementing the ability to open the backpack while wearing it or holding it in hand be considered?

**Comment by BeansGalaxy** [comment]
I won't add opening the backpacks with a hand since that's how every other backpack mod does it and a player couldn't be stopped from having 36 backpacks which that's what shulkers are good for. I think needing to place or equip the backpack to interact with the items and being limited to one while you travel is what makes my mod so unique. 

Other than that what do you mean by templates?

**Reply by DareckGobos** [reply]
I respect that it is a mark of the mod, I forgot to suggest a modification regarding the functionality of the backpack in the game. I recognize that the ability to inspect the backpack from the outside is well implemented, but I consider breaking the backpack and dropping all the items to be an extreme measure. Instead, I suggest that the backpack retain its contents even when torn, to preserve the comfort and design of this particular item. You also mentioned the possibility of allowing multiple backpacks to be carried but with other mods they implement a mechanic that disadvantages the player, although they recognize that this may not be intuitive, but they consider it appropriate for this specific case. Lastly, when it comes to "templates" (blacksmith molds) I mean the ability to edit them as you do with armor.

**Reply by BeansGalaxy** [reply]
The backpacks are coded so that it's contents are stored on the player and not in the actual backpack item when equipped. I have just changed this in the case of wearing cauldrons & pots which their items do actually store their contents but in the case of backpacks, there's just too much information there to store it in the item. 

Because of that, breaking the backpack with a punch has to drop it's items because there's no way to store them yet. In 1.21 they are making HUGE changes to nbt and I'll be saving the contents to the actual item there since it'll be more efficient that way in that version. 

And templates can be used on metal backpacks, not too sure what you mean when you say they could enhance the appeal since it's already in the game. Unless you mean trimming leather backpacks which I plan to keep dying and trimming separate because I like how it makes metal backpacks feel less like an upgrade and more like a variant.

**Reply by DareckGobos** [reply]
I meant that aesthetically it looked good to edit the backpacks with these templates, for that aspect I don't have any suggestions, it's very good. 
On the other hand, I'm still looking forward to trying your backpack with its gears until you move it to 1.21, but I understand the reasons because they were just suggestions, it was a pleasure to talk.
